### PR TITLE
Update to Quarkus 3.7

### PR DESCRIPTION
**Pull request by gsmet**
I wasn't able to test it as tests require a token but I thought it might help anyway.

### CODE REVIEW

File: runtime/src/main/java/io/quarkiverse/discord4j/runtime/health/Discord4jHealthCheck.java [hunk 1]
```diff
@@ -38,16 +39,15 @@ public Uni<HealthCheckResponse> call() {
...
 
-            unis.add(
-                    Uni.createFrom().publisher(client.get().isConnected().doOnNext(connected -> {
-                        if (connected) {
-                            builder.withData(shardName + ".response.time", responseTime);
-                        } else {
-                            builder.down().withData("reason", shardName + " is not connected");
-                        }
-                    })));
+            unis.add(UniReactorConverters.<Boolean> fromMono().from(client.get().isConnected().doOnNext(connected -> {
```
Comment (gsmet) [inline runtime/src/main/java/io/quarkiverse/discord4j/runtime/health/Discord4jHealthCheck.java:42]: @jponge this is not pretty but I couldn't find a better way to achieve this?

**Comment by gsmet** [comment]
I was able to set up CI with a test Discord server.

**Comment by gsmet** [comment]
I'm going to merge this as it's a step in the right direction.

File: runtime/src/main/java/io/quarkiverse/discord4j/runtime/health/Discord4jHealthCheck.java (hunk 1 above)
Comment (jponge) [reply runtime/src/main/java/io/quarkiverse/discord4j/runtime/health/Discord4jHealthCheck.java:42]: No that's what you need, aka adapt a legacy API `Publisher` to a JDK `Flow.Publisher`

**Pull request merged**